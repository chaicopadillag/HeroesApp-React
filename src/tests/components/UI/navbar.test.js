const { mount } = require('enzyme');
const { MemoryRouter, Router } = require('react-router-dom');
const { AuthContext } = require('../../../auth/AuthContext');
const { Navbar } = require('../../../components/UI/Navbar');
const { types } = require('../../../types/types');

describe('Prueba de componente <Navbar/>', () => {
	const historyMock = {
		push: jest.fn(),
		replace: jest.fn(),
		location: jest.fn(),
		listen: jest.fn(),
		createHref: jest.fn(),
	};
	const contextvalue = {
		dispatch: jest.fn(),
		session: {
			nombre: 'Code Codero',
			correo: 'codecodero@gmail.com',
			sesion: true,
		},
	};
	const wrapper = mount(
		<AuthContext.Provider value={contextvalue}>
			<MemoryRouter>
				<Router history={historyMock}>
					<Navbar />
				</Router>
			</MemoryRouter>
		</AuthContext.Provider>
	);
	afterEach(() => {
		jest.clearAllMocks();
	});
	test('Debe mostrarse correctamente', () => {
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('.text-warning').text().trim()).toBe('Code Codero');
	});
	test('Debe de llamar el logout y history', () => {
		wrapper.find('button').simulate('click');
		expect(contextvalue.dispatch).toHaveBeenCalledWith({ type: types.logout });
		expect(historyMock.replace).toHaveBeenCalledWith('/login');
	});
});
