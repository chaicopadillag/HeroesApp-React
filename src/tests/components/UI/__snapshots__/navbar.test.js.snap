// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente <Navbar/> Debe mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      onClick={[Function]}
    >
      Heroes App
    </a>
    <div
      className="collapse navbar-collapse justify-content-between"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
      <div
        className="d-flex"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <span
            className="nav-item nav-link text-warning"
          >
            Code Codero
          </span>
          <button
            className="btn btn-outline-warning"
            onClick={[Function]}
          >
            Logout
          </button>
        </ul>
      </div>
    </div>
  </div>
</nav>
`;
